var/spack/repos/builtin/packages/openfoam/package.py

```diff
@@ -0,0 +1,358 @@
+##############################################################################
+# Copyright (c) 2017 OpenCFD Ltd.
+#
+# This file is released as part of Spack.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for the LLNL notice and the LGPL.
+#
+# License
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+from spack.environment import *
+
+import multiprocessing
+import glob
+import os
+import re
+import shutil
+
+
+# Notes.
+# - Resolution of flex, zlib needs more attention
+# - Paraview/VTK build - needs adjustments in ~/.spack/packages.yaml
+class Openfoam(Package):
```

Comment:
Understood, but I'm not sure how long the '+' part of plus will actually be around (both OpenFOAM and OpenFOAM+ are in use at OpenCFD). The main intention of the '+' was to distinguish from the 3.0 .org version, which dropped many features that were given to them for release. For clarity and to remain neutral, either we keep openfoam as a package name and supply +com and +org variants (eventually adding in +extend in the process), or we spit things entirely and have openfoam-com, openfoam-org, foam-extend as packages, each providing 'openfoam' as a virtual (similar to mpi treatment). This would also allow other combinations without any further problems. Additional packages like swak4Foam that build with several different versions could require 'openfoam' and then be able to build various variants. This sounds to me to be the cleanest solution.